assets/js/blocks/filter-utils.tsx

```diff
@@ -0,0 +1,25 @@
+export const mapRawQueryToUseCollection = ( rawQuery ) => {
```

Comment:
Hey @kmanijak, Thank you so much for working on this 🙏🏻

I have a similar concern as you have. It appears we are manually mapping all fields, which could potentially raise some issues down the line. Here are some of my thoughts:
- **Backend query format changes**: If modification occur within PHP files, altering the format of our queries, our current solution may not adapt well to these changes. This could lead to potential discrepancies and increased maintenance efforts. 
- **Store API Format changes**: Similiarly, if changes are made to the format of store API, we may encounter compatibility issues. This could further complicate our code maintenance and update process.
- **Addition of Fields in Future**: If we plan to include more fields in the PHP files in the future, this might necessitate further manual mapping. This approach could incrementally increase the complexity of our solution, making it less efficient and more prone to errors.

My original assumption was that we would be able to avoid significant manual mapping and the solution would largely work out of the box.

What are your thoughts? Thanks again for working on this 🙂

Reply:
> My original assumption was that we would be able to avoid significant manual mapping and the solution would largely work out of the box.

TBH I was hoping the same. Unfortunately, the data formats are incompatible and what's worse: there are differences between in the backed query between Products and Products Collection. I suspect some bug with Products block considering it doesn't work well now with the filter counters.

But yes, maintaining this layer can be costly, I agree!